searx/engines/docker_hub.py

```diff
@@ -19,14 +19,17 @@
 categories = ['it', 'packages']  # optional
 paging = True
 
-base_url = "https://hub.docker.com/"
-search_url = base_url + "api/content/v1/products/search?{query}&type=image&page_size=25"
+base_url = "https://hub.docker.com"
+page_size = 10
 
 
 def request(query, params):
-
-    params['url'] = search_url.format(query=urlencode(dict(q=query, page=params["pageno"])))
-    params["headers"]["Search-Version"] = "v3"
+    args = {
+        "query": query,
+        "from": page_size * (params['pageno'] - 1),
+        "size": page_size,
+    }
+    params['url'] = f"{base_url}/api/search/v3/catalog/search?{urlencode(args)}"
```

Comment:
The the link from line 3 ..

```
"official_api_documentation": 'https://docs.docker.com/registry/spec/api/',
```
.. seems no longer valid .. from where did you got the `/api/search/v3/catalog/search`?

Reply:
Via their Web UI, just as you probably, I can't seem to find any documentation about the catalog API.

Nevertheless it might make sense to keep the documentation link there in case they update their documentation? Their website claims that the API is still in beta, so that might be why it's not yet fully documented.